vhd/lib/libvhd.c

```diff
@@ -1509,11 +1509,15 @@ vhd_macx_decode_location(char *in, char *out, int len)
 	ibl  = obl = len;
 
 	cd = iconv_open("ASCII", "UTF-8");
-	if (cd == (iconv_t)-1) 
+	if (cd == (iconv_t)-1) {
+		iconv_close(cd);
```

Comment:
This isn't required and in fact will fail. Only the chunk below is necessary. If `iconv_open` open fails then cd does not refer to an open conversion descriptor.